src/pyodide/internal/process_script_imports.py

```diff
@@ -23,12 +29,20 @@ def process_script(script):
         for mod in find_imports(script):
             try:
                 __import__(mod)
+                CF_LOADED_MODULES.append(mod)
             except ImportError:
                 pass
 
     def process_scripts():
-        for script in Path("/session/metadata").glob("**/*.py"):
+        # Currently this script assumes that it is generating a _package_ snapshot- one that
+        # only includes non-vendored packages. Because of this we do not wish to import local
+        # modules, the easiest way to ensure they cannot be imported is to remove
+        # `/session/metadata` from the sys path.
```

Comment:
Sure, that makes sense. I wonder should we instead delay putting `"/session/metadata"` on the path until after this code? But I guess it doesn't make too much of a difference.

Reply:
The advantage of doing it here is that it ensures the `/session/metadata` path is not there right before the imports are done, so it's less likely we'll accidentally include it in the path.